envi/tests/test_arch_arm.py

```diff
@@ -5,68 +5,591 @@
+        (REV_ALL, '6745a3e0', 0x4560, 'adc r4, r3, r7, ror #10', 0, ()),
+        (REV_ALL, '6745b3e0', 0x4560, 'adcs r4, r3, r7, ror #10', 0, ()),
+        (REV_ALL, '6745c3e0', 0x4560, 'sbc r4, r3, r7, ror #10', 0, ()),
+        (REV_ALL, '6745d3e0', 0x4560, 'sbcs r4, r3, r7, ror #10', 0, ()),
+        (REV_ALL, '6745e3e0', 0x4560, 'rsc r4, r3, r7, ror #10', 0, ()),
+        (REV_ALL, '6745f3e0', 0x4560, 'rscs r4, r3, r7, ror #10', 0, ()) 
+        ] 
+
 # FIXME: create list of this from IDA flow below - THIS CURRENT DATA IS FOR H8!  NOT ARM/THUMB
+        
 '''
-        ( '8342', 0x4560, 'add.b #42, r3h', IF_B, () ),
-        ( '7c6075f0', 0x4560, 'bixor #7, @er6', 0, () ),
-        ( '7d507170', 0x4560, 'bnot #7, @er5', 0, () ),
-        ( '0832', 0x4560, 'add.b r3h, r2h', IF_B, () ),
-        ( '791d4745', 0x4560, 'add.w #4745, e5', IF_W, () ),
-        ( '0932', 0x4560, 'add.w r3, r2', IF_W, () ),
-        ( '7a1d00047145', 0x4560, 'add.l #47145, er5', IF_L, () ),
-        ( '01406930', 0x4560, 'ldc.w @er3, ccr', IF_W, () ),
-        ( '014069b0', 0x4560, 'stc.w ccr, @er3', IF_W, () ),
-        ( '01c05023', 0x4560, 'mulxs.b r2h, r3', IF_B, () ),
-        ( '01c05223', 0x4560, 'mulxs.w r2, er3', IF_W, () ),
-        ( '01d05123', 0x4560, 'divxs.b r2h, r3', IF_B, () ),
-        ( '01d05323', 0x4560, 'divxs.w r2, er3', IF_W, () ),
-        ( '01f06423', 0x4560, 'or.l er2, er3', IF_L, () ),
-        ( '01f06523', 0x4560, 'xor.l er2, er3', IF_L, () ),
-        ( '01f06623', 0x4560, 'and.l er2, er3', IF_L, () ),
-        ( '0a03', 0x4560, 'inc.b r3h', IF_B, () ),
-        ( '0a83', 0x4560, 'add.l er0, er3', IF_L, () ),
-        ( '0b83', 0x4560, 'adds #2, er3', 0, () ),
-        ( '0b93', 0x4560, 'adds #4, er3', 0, () ),
-        ( '0b53', 0x4560, 'inc.w #1, r3', IF_W, () ),
-        ( '0bf3', 0x4560, 'inc.l #2, er3', IF_L, () ),
-        ( '0f00', 0x4560, 'daa r0h', 0, () ),
-        ( '0f93', 0x4560, 'mov.l er1, er3', IF_L, () ),
-        ( '1a03', 0x4560, 'dec.b r3h', IF_B, () ),
+        leaving here for reference.
+        will remove this section once the emulation test code is started
+        Currently emulation is disabled until disassembly is all done
         ( '1a83', 0x4560, 'sub.l er0, er3', IF_L, (
             {'setup':(('er0',0xaa),('CCR_C',0),('er3',0x1a)), 
                 'tests':(('er3',0x90),('CCR_H',0),('CCR_N',0),('CCR_Z',0),('CCR_V',0),('CCR_C',0)) },
             {'setup':(('er0',0xab),('CCR_C',0),('er3',0xb0)), 
                 'tests':(('er3',0xfffffffb),('CCR_H',1),('CCR_N',1),('CCR_Z',0),('CCR_V',0),('CCR_C',1)) },
             ) ),
-        ( '1b83', 0x4560, 'subs #2, er3', 0, () ),
-        ( '1b93', 0x4560, 'subs #4, er3', 0, () ),
-        ( '1b53', 0x4560, 'dec.w #1, r3', IF_W, () ),
-        ( '1bf3', 0x4560, 'dec.l #2, er3', IF_L, () ),
-        '''
+'''
 
 # temp scratch: generated these while testing
 ['0de803c0','8de903c0','ade903c0','2de803c0','1de803c0','3de803c0','9de903c0','bde903c0',]
```

Comment:
I haven't started looking at the thumb code yet. Figured on working through the v7 code first then approaching the Thumb code. 

Can you do an IDA dump of thumb2 codes? 
I can easily convert the dump into tests by modifying my helper program.
But without disassembling the code myself by hand I am not sure how to create the full command.
Assuming you are talking about the #temp scratch stuff

Date: Tue, 7 Jun 2016 10:54:24 -0700
From: notifications@github.com
To: vivisect@noreply.github.com
CC: redheadedrod@hotmail.com; author@noreply.github.com
Subject: Re: [atlas0fd00m/vivisect] Unittest (#5)

In envi/tests/test_arch_arm.py:

>          ( '1a83', 0x4560, 'sub.l er0, er3', IF_L, (
>              {'setup':(('er0',0xaa),('CCR_C',0),('er3',0x1a)), 
>                  'tests':(('er3',0x90),('CCR_H',0),('CCR_N',0),('CCR_Z',0),('CCR_V',0),('CCR_C',0)) },
>              {'setup':(('er0',0xab),('CCR_C',0),('er3',0xb0)), 
>                  'tests':(('er3',0xfffffffb),('CCR_H',1),('CCR_N',1),('CCR_Z',0),('CCR_V',0),('CCR_C',1)) },
>              ) ),
> -        ( '1b83', 0x4560, 'subs #2, er3', 0, () ),
> -        ( '1b93', 0x4560, 'subs #4, er3', 0, () ),
> -        ( '1b53', 0x4560, 'dec.w #1, r3', IF_W, () ),
> -        ( '1bf3', 0x4560, 'dec.l #2, er3', IF_L, () ),
> -        '''
> +'''
>  
>  # temp scratch: generated these while testing
>  ['0de803c0','8de903c0','ade903c0','2de803c0','1de803c0','3de803c0','9de903c0','bde903c0',]

note:  these are byte sequences which translate to thumb2 opcodes, and should be turned into test cases :)


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.